common-custom-user-data-maven-extension/src/main/java/com/gradle/CommonCustomUserDataMavenExtension.java

```diff
@@ -31,6 +32,13 @@ public CommonCustomUserDataMavenExtension(PlexusContainer container, Logger logg
     public void afterProjectsRead(MavenSession session) throws MavenExecutionException {
         logger.debug("Executing extension: " + getClass().getSimpleName());
 
+        GradleEnterpriseApi gradleEnterprise = ApiAccessor.lookupGradleEnterpriseApi(container, getClass());
+        if (gradleEnterprise != null) {
+            logger.debug("Configuring Gradle Enterprise");
+            CustomGradleEnterpriseConfig.configureGradleEnterprise(gradleEnterprise);
+            logger.debug("Finished configuring Gradle Enterprise");
+        }
+
         BuildScanApi buildScan = ApiAccessor.lookupBuildScanApi(container, getClass());
```

Comment:
I think we should remove these additional lookups. @etiennestuder Do you think we should keep the separate methods in `CustomGradleEnterpriseConfig`?

Reply:
Hm, I think I would keep it as it is. But I can take another look when I merge it.

What is your rationale for changing it?

Reply:
It would emphasize that there's a single entrypoint but I'm okay with leaving it as-is.

Reply:
If you remove it, then it breaks existing customer scripts

Reply:
I wasn't suggesting to remove it from the Groovy script support, just from the Java "templates".

Reply:
I now see what you mean, @marcphilipp . I'll adjust the code accordingly. Thanks.

Reply:
Fixed in this [commit](https://github.com/gradle/gradle-enterprise-build-config-samples/commit/58f49dbaceb10a3500c24c77b6673b6dd41530f9).